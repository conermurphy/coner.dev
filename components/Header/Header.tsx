import React from 'react';
import NavBar from '../NavBar/NavBar';

export default function Header(): JSX.Element {
  return (
    <div className="sticky top-0 z-20">
      <header className="flex flex-row items-center justify-between h-16 px-106 bg-transparent">
        <div className="flex flex-row items-center gap-12">
          <p className="text-2xl font-bold opacity-100">Coner Murphy</p>
          <NavBar />
        </div>
      </header>
    </div>
  );
}
