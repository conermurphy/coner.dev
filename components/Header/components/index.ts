export { default as HeaderBackground } from './HeaderBackground/HeaderBackground';
