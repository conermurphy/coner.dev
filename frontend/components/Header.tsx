import React, { useState } from 'react';
import Link from 'next/link';
import { Dialog } from '@headlessui/react';
import Logo from './Logo/Logo';
import { getIcon } from '../utils';
import { ICONS, NAVIGATION } from '../constants';
import Socials from './Socials/Socials';

export default function Header() {
  const [mobileMenuOpen, setMobileMenuOpen] = useState(false);

  return (
    <header className="bg-background font-heading">
      <nav
        className="mx-auto flex max-w-7xl items-center justify-between p-6 lg:px-8"
        aria-label="Global"
      >
        <Link href="/" className="w-12 h-12 flex items-center p-1.5">
          <span className="sr-only">Coner Murphy</span>
          <Logo />
        </Link>
        <div className="flex lg:hidden">
          <button
            type="button"
            className="-m-2.5 inline-flex items-center justify-center rounded-md p-2.5 text-text/90"
            onClick={() => setMobileMenuOpen(true)}
          >
            <span className="sr-only">Open main menu</span>
            {getIcon({ icon: ICONS.JUSTIFY.name, size: '24px' })}
          </button>
        </div>
        <div className="hidden lg:flex lg:gap-x-12">
          {NAVIGATION.map((item) => (
            <Link
              key={item.name}
              href={item.href}
              className="text-lg font-bold leading-6 text-text/90 border-b-2 border-transparent hover:border-brand border-solid w-max"
            >
              {item.name}
            </Link>
          ))}
        </div>
      </nav>
      <Dialog
        as="div"
        className="lg:hidden"
        open={mobileMenuOpen}
        onClose={setMobileMenuOpen}
      >
        <div className="fixed inset-0 z-60" />
        <Dialog.Panel className="fixed inset-y-0 right-0 z-10 w-full overflow-y-auto bg-background px-6 py-6 sm:max-w-sm sm:ring-1 sm:ring-gray-900/10 drop-shadow-md">
          <div className="flex items-center justify-between">
            <Link
              href="/"
              onClick={() => setMobileMenuOpen(false)}
              className="w-12 h-12 p-1.5 flex items-center"
            >
              <span className="sr-only">Coner Murphy</span>
              <Logo />
            </Link>
            <button
              type="button"
              className="-m-2.5 rounded-md p-2.5 text-text/90"
              onClick={() => setMobileMenuOpen(false)}
            >
              <span className="sr-only">Close menu</span>
              {getIcon({ icon: ICONS.CLOSE.name, size: '24px' })}
            </button>
          </div>
          <div className="mt-6 flex flex-col gap-6">
            <div className="-my-6 divide-y divide-gray-500/10">
              <div className="space-y-2 py-6 px-3">
                {NAVIGATION.map((item) => (
                  <a
                    key={item.name}
                    href={item.href}
                    className="-mx-3 block py-2 text-base font-semibold leading-7 text-text/90 border-transparent border-b-2 hover:border-brand border-solid w-max"
                  >
                    {item.name}
                  </a>
                ))}
              </div>
            </div>
            <Socials />
          </div>
        </Dialog.Panel>
      </Dialog>
    </header>
  );
}
