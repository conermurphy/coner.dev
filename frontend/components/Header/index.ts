export { default as DesktopHeader } from './DesktopHeader/DesktopHeader';
export { default as MobileHeader } from './MobileHeader/MobileHeader';
