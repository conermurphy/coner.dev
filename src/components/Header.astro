---
import HeaderLink from './HeaderLink.astro'
import { SITE_TITLE } from '../consts'
import { Image } from 'astro:assets'
import Logo from '../images/logo.svg'
---

<header>
  <nav
    class="fixed left-1/2 top-4 z-[90] flex w-max -translate-x-1/2 items-center justify-between gap-x-10 rounded-2xl bg-light/10 px-4 py-4 backdrop-blur-md sm:max-w-2xl lg:justify-center lg:px-10"
    aria-label="Global"
  >
    <div class="hidden lg:flex lg:gap-x-12">
      <HeaderLink href="/">Home</HeaderLink>
      <HeaderLink href="/contact">Contact Me</HeaderLink>
    </div>

    <a href="/" class="-m-1.5 p-1.5">
      <span class="sr-only">{SITE_TITLE}</span>
      <Image class="h-5 w-auto" src={Logo} alt="" />
    </a>

    <div class="flex items-center lg:hidden">
      <button
        type="button"
        class="-m-2.5 inline-flex items-center justify-center rounded-md p-2.5 text-light"
      >
        <span class="sr-only">Open main menu</span>
        <svg
          class="size-6"
          fill="none"
          viewBox="0 0 24 24"
          stroke-width="1.5"
          stroke="currentColor"
          aria-hidden="true"
          data-slot="icon"
        >
          <path
            stroke-linecap="round"
            stroke-linejoin="round"
            d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"></path>
        </svg>
      </button>
    </div>

    <div class="hidden lg:flex lg:gap-x-12">
      <HeaderLink href="/blog">Blog</HeaderLink>
      <HeaderLink href="/technical-writing">Technical Writing</HeaderLink>
    </div>
  </nav>

  <!-- Mobile menu, show/hide based on menu open state. -->
  <div class="group lg:hidden" role="dialog" aria-modal="true">
    <!-- Background backdrop, show/hide based on slide-over state. -->
    <div
      class="fixed inset-0 -z-10 transition duration-300 ease-in-out group-aria-expanded:z-[90] group-aria-expanded:backdrop-blur-sm"
    >
    </div>
    <div
      class="fixed inset-y-0 right-0 z-[999] w-full max-w-60 translate-x-60 overflow-y-auto bg-dark px-10 py-6 transition-transform duration-150 ease-in-out group-aria-expanded:translate-x-0"
    >
      <div class="flex w-full items-center justify-between">
        <a href="/" class="-m-1.5 p-1.5">
          <span class="sr-only">{SITE_TITLE}</span>
          <Image class="h-8 w-auto" src={Logo} alt="" />
        </a>

        <div>
          <button type="button" class="-m-2.5 rounded-md p-2.5 text-light">
            <span class="sr-only">Close menu</span>
            <svg
              class="size-6"
              fill="none"
              viewBox="0 0 24 24"
              stroke-width="1.5"
              stroke="currentColor"
              aria-hidden="true"
              data-slot="icon"
            >
              <path
                stroke-linecap="round"
                stroke-linejoin="round"
                d="M6 18 18 6M6 6l12 12"></path>
            </svg>
          </button>
        </div>
      </div>

      <div class="mt-6 flex flex-col space-y-2">
        <HeaderLink href="/">Home</HeaderLink>
        <HeaderLink href="/blog">Blog</HeaderLink>
        <HeaderLink href="/technical-writing">Technical Writing</HeaderLink>
        <HeaderLink href="/contact">Contact Me</HeaderLink>
      </div>
    </div>
  </div>
</header>

<script>
  const mobileMenuContainer = document.querySelector('[role="dialog"]')
  const buttons = document.querySelectorAll('button')

  buttons.forEach((button) => {
    button.addEventListener('click', () => {
      const isExpanded =
        mobileMenuContainer?.getAttribute('aria-expanded') === 'true'
      mobileMenuContainer?.setAttribute('aria-expanded', `${!isExpanded}`)
    })
  })
</script>
