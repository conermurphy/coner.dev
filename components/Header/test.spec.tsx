import { render, screen } from '@testing-library/react';
import React from 'react';
import Header from './Header';

describe('Header', () => {
  it('Should render the site title', () => {
    render(<Header />);

    const title = screen.getByText(/coner murphy/i);

    expect(title).toBeVisible();
  });

  it('Should render a link to home', () => {
    render(<Header />);

    const title = screen.getByRole('link', { name: 'Coner Murphy' });

    expect(title).toBeVisible();
    expect(title.getAttribute('href')).toMatch(/.*\/$/);
  });

  it('Should render a nav component for the nav bar', () => {
    render(<Header />);

    const nav = screen.getByRole('navigation');

    expect(nav).toBeVisible();
  });
});
