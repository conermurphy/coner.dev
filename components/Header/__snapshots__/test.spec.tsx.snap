// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Matches the snapshot with no props 1`] = `
<div>
  <div
    class="sticky top-0 z-20"
  >
    <header
      class="flex flex-row items-center justify-between h-16 px-106 bg-transparent"
    >
      <div
        class="flex flex-row items-center gap-12"
      >
        <p
          class="text-2xl font-bold opacity-100"
        >
          Coner Murphy
        </p>
        <nav
          class="flex flex-row gap-8"
        >
          <a
            class="font-semibold"
            href="/"
          >
            Home
          </a>
          <a
            class="font-semibold"
            href="/blog"
          >
            Blog
          </a>
          <a
            class="font-semibold"
            href="/contact"
          >
            Contact
          </a>
        </nav>
      </div>
    </header>
  </div>
</div>
`;
